tmt/steps/report/polarion.py

```diff
@@ -224,53 +232,46 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
             'description', 'planned_in', 'assignee', 'pool_team', 'arch', 'platform', 'build',
             'sample_image', 'logs', 'compose_id', 'fips']
 
-        xml_data = make_junit_xml(
-            phase=self,
+        # TODO: The current behavior is that the None value is propagated into the final XML as a
```

Comment:
@KwisatzHaderach
What is the expected behavior here? Should the `None` value always be propagated into the property value as a string `"None"? My expectation would be the `None` should propagate as an empty string.

Reply:
project_id is required, I'm quite surprised I've forgot to add a check for it and raise in case it's missing

so my suggestion would be to add a last resort project_id taken from the login credentials `PolarionWorkItem._session.default_project` and then raising if still empty/None (AFAIK the argument is required for pylero so it shouldn't happen), only bad detail is this can lead to unexpected reports in whatever the user has set as their default project, but at least it would be whatever they use in authentication which should be fine

Reply:
Thanks for the reply. I've added value from `PolarionWorkItem._session.default_project` as a last resort and I've tried to document it. If the `bool(project_id)` still evaluates to `False`, I am now raising a `ReportError` exception.

I also added the test which checks there is no `None` string in the property value:

```
rlAssertNotGrep 'value="None"' "xunit.xml"
```

Reply:
yeah, that sounds like a good idea